### PR TITLE
feat: create parse script data method

### PR DESCRIPTION
**Pull request by pedroferreira1**
### Acceptance Criteria
- We must have a method to parse an output script that is a script data.

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/utils/scripts.ts [hunk 1]
```diff
@@ -37,3 +39,87 @@ export const parseP2PKH = (buff: Buffer, network: Network): P2PKH => {
...
+  expectedLen += dataBytesLen;
+
+  if (expectedLen !== scriptBuf.length) {
+    // The script has different qty of bytes than expected
+    throw new ParseScriptError('Invalid output script.');
+  }
+
+  if (scriptBuf[expectedLen - 1] !== OP_CHECKSIG[0]) {
+    // Last byte must be an OP_CHECKSIG
+    throw new ParseScriptError('Invalid output script.');
```
Comment (luislhl) [inline src/utils/scripts.ts:83]: Should the error message better describe what is happening in each case?
Like: `Invalid output script. Last byte must be an OP_CHECKSIG`.

**Review by andreabadesso — APPROVED** [review]

File: __tests__/models/output.test.js [hunk 2]
```diff
@@ -97,4 +98,24 @@ test('Script', () => {
   expect(() => {
     parseP2PKH(o1.script.slice(10), network);
   }).toThrowError(ParseScriptError);
+});
+
+test('Script data', () => {
```
Comment (luislhl) [inline __tests__/models/output.test.js:103]: Shouldn't this test be included in a new file `__tests__/models/script_data.test.js`?

If I understand the naming pattern correctly, each model should have its own test file, right?

File: src/utils/scripts.ts [hunk 3]
```diff
@@ -37,3 +39,87 @@ export const parseP2PKH = (buff: Buffer, network: Network): P2PKH => {
...
+  let scriptBuf = _.clone(buff);
+
+  if (scriptBuf.length === 0) {
+    throw new ParseError('Invalid buffer.');
+  }
+
+  let lenData, start;
+
+  if (scriptBuf[0] > 75) {
+    lenData = scriptBuf[1];
```
Comment (luislhl) [inline src/utils/scripts.ts:118]: Should we have a test for this case?

File: src/utils/scripts.ts [hunk 4]
```diff
@@ -37,3 +39,87 @@ export const parseP2PKH = (buff: Buffer, network: Network): P2PKH => {
...
+  // The expected len will be at least 2 bytes
+  // 1 for the script len and 1 for the OP_CHECKSIG in the end
+  let expectedLen = 2;
+  let dataBytesLen;
+
+  // If we have OP_PUSHDATA1 as first byte, the second byte has the length of data
+  // otherwise, the first byte already has the length of data
+  if (scriptBuf[0] === OP_PUSHDATA1[0]) {
+    expectedLen += 1;
+    dataBytesLen = scriptBuf[1];
```
Comment (luislhl) [inline src/utils/scripts.ts:68]: Should we have a test for this case?

Comment (pedroferreira1) [reply src/utils/scripts.ts:68]: Agreed. Added here https://github.com/HathorNetwork/hathor-wallet-lib/pull/295/commits/b7bf56105ac7290106987153ba64e1e302821ee5

File: src/utils/scripts.ts (hunk 3 above)
Comment (pedroferreira1) [reply src/utils/scripts.ts:118]: Agreed. Added here https://github.com/HathorNetwork/hathor-wallet-lib/pull/295/commits/b7bf56105ac7290106987153ba64e1e302821ee5

File: src/utils/scripts.ts (hunk 1 above)
Comment (pedroferreira1) [reply src/utils/scripts.ts]: Makes sense. Added more details in all error messages here https://github.com/HathorNetwork/hathor-wallet-lib/pull/295/commits/8d6c46cfdf5d96b377f4a33118805f191095380f

**Review by luislhl — APPROVED** [review]

**Pull request merged**